micro_sam/prompt_generators.py

```diff
@@ -252,64 +252,76 @@ def __call__(
 class IterativePromptGenerator(PromptGeneratorBase):
     """Generate point prompts from an instance segmentation iteratively.
     """
-    def _get_positive_points(self, pos_region, overlap_region):
+    def _get_positive_points(self, pos_region, overlap_region, is_3d):
         positive_locations = [torch.where(pos_reg) for pos_reg in pos_region]
         # we may have objects without a positive region (= missing true foreground)
-        # in this case we just sample a point where the model was already correct
+        # in this case we just sample a positive point where the model was already correct
         positive_locations = [
             torch.where(ovlp_reg) if len(pos_loc[0]) == 0 else pos_loc
             for pos_loc, ovlp_reg in zip(positive_locations, overlap_region)
         ]
-        # we sample one location for each object in the batch
+        # we sample one positive location for each object in the batch
         sampled_indices = [np.random.choice(len(pos_loc[0])) for pos_loc in positive_locations]
-        # get the corresponding coordinates (Note that we flip the axis order here due to the expected order of SAM)
-        pos_coordinates = [
-            [pos_loc[-1][idx], pos_loc[-2][idx]] for pos_loc, idx in zip(positive_locations, sampled_indices)
-        ]
+        # get the corresponding coordinates (NOTE: we flip the axis order here due to the expected order of SAM)
+        pos_coordinates = []
```

Comment:
This is a minor comment, but you should try to keep list comprehensions whenever possible since they are faster. 
That is quite easy here. you can do:
```python
if is_3d:
   pos_coordinates = [....]
else:
  pos_coordinates = [...]
```
instead of getting rid of the list comprehension and writing out the full loop.

Reply:
Okay, I'll keep that into account. This should be taken care of now. Thanks for the mention!